packages/typespec-client-generator-core/src/package.ts

```diff
@@ -573,7 +573,6 @@ function createSdkClientType<
     // eslint-disable-next-line deprecation/deprecation
     arm: client.kind === "SdkClient" ? client.arm : false,
   };
-  context.__clients!.push(sdkClientType);
```

Comment:
curious why do we delete this?

Reply:
it is useless to cache clients since we only expose top level in the sdkpacakge.